api/migrations/20190305160606_add_sf_type_and_version.sql

```diff
@@ -0,0 +1,16 @@
+
+-- +goose Up
+-- SQL in section 'Up' is executed when this migration is applied
+-- +goose StatementBegin
+ALTER TABLE accounts ADD COLUMN sf_type VARCHAR(25) NOT NULL DEFAULT 'SF86';
```

Comment:
I think we need to leave the backend be the arbiter of what form gets assigned and not have it spread across multiple components. For now, the backend is going to assign this default, but eventually it might come over from an external service request.

Reply:
To be clear – I would remove the defaults from the table schema and just have default logic at the app layer.

Reply:
The update script should set the column values for any existing rows to be SF-86, 2016-11.

Reply:
Having the default in the table logic is the only way to have NOT NULL work at the table layer. The current DB has almost no db level validation logic and I think that's a mistake. I figured this would be a good place to start experimenting with enforcing some db requirements. 

I can undo that and have the startup script set things instead and do all validation logic in the backend layer instead but I prefer this for now.

Eventually, we are going to have to have a "forms" table and this data can live there instead of on the account. But I believe that the form type is passed in during account provisioning so I think that this not null requirement can be enforced for now.

Reply:
Yeah, we are on the same page in getting more constraints in the db – we should keep the not null. 

But in the update script, you should be able to add the columns, populate them with the defaults, then add the not null constraint after that.  You could even define these columns as an enum type that would constrain the value range to the map you have defined in the app layer.

To be clear, I'm just saying that the decision as to what is the default form type should solely be made at the app layer.